### PR TITLE
add `cupy.random.exponential`

### PR DESCRIPTION
**Pull request by YoshikawaMasashi**
There is error in `cp.random.standard_exponential`. I fixed.

### CODE REVIEW

**Comment by YoshikawaMasashi** [comment]
Jenkins, test this please.

File: tests/cupy_tests/random_tests/test_distributions.py [hunk 1]
```diff
@@ -99,6 +99,22 @@ def test_dirichlet(self, alpha_dtype, dtype):
...
+@testing.parameterize(*testing.product({
+    'shape': [(4, 3, 2), (3, 2)],
+    'scale_shape': [(), (3, 2)],
+})
+)
+@testing.gpu
+class TestDistributionsExponential(RandomDistributionsTestCase):
+
+    @cupy.testing.for_float_dtypes('dtype', no_float16=True)
+    @cupy.testing.for_float_dtypes('scale_dtype')
```
Comment (toslunar) [inline tests/cupy_tests/random_tests/test_distributions.py]: `numpy.random.exponential` does not support complex dtypes:
```
>>> numpy.random.exponential(scale=numpy.array([1, 2j]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 1769, in mtrand.RandomState.exponential
TypeError: Cannot cast array data from dtype('complex128') to dtype('float64') according to the rule 'safe'
```

Comment (toslunar) [reply tests/cupy_tests/random_tests/test_distributions.py]: Sorry.  You may test integer dtypes but you don't have to.

File: cupy/random/generator.py [hunk 2]
```diff
@@ -135,6 +135,16 @@ def dirichlet(self, alpha, size=None, dtype=float):
...
         return y
 
+    def exponential(self, scale=1.0, size=None, dtype=float):
+        """Returns an array of samples drawn from a exponential distribution.
+
+        .. seealso::
+            :func:`cupy.random.exponential` for full documentation,
+            :meth:`numpy.random.RandomState.exponential`
+        """
+        x = self.standard_exponential(size, dtype)
```
Comment (toslunar) [inline cupy/random/generator.py]: Update `size` using `scale.shape` before calling `standard_exponential`.  The following is failing.
```
cupy.random.exponential(cupy.array([1., 2.]), size=None)
```

**Comment by toslunar** [comment]
Merge #1642 first.

**Comment by toslunar** [comment]
It's OK to be fixed in another PR.
Jenkins, test this please.

**Pull request merged**